python-test/features/steps/functions.py

```diff
@@ -0,0 +1,84 @@
+import requests
+import json
+import os
+import docker
+import re
+import hamcrest
+
+base_orb_url = "https://" + os.getenv('ORB_ADDRESS', "beta.orb.live")
+
+def generate_token(email, password, base_orb_url = base_orb_url):
+    if email != None and password != None: 
+        headers={'Content-type':'application/json',  'Accept':'*/*'}
+        token_request = requests.post(base_orb_url+'/api/v1/tokens', json= {'email': email, 'password': password}, headers=headers)
+        hamcrest.assert_that(token_request.json(), hamcrest.has_key('token'), 'Request Json - token')
+        return token_request.json()
+    else:
+        hamcrest.assert_that(email, hamcrest.not_none(), 'Email')
+        hamcrest.assert_that(password, hamcrest.not_none(), 'Password')
```

Comment:
Why not asserting before the if/else block? In fact, if we do that we won't need the if/else in the first place.